src/navbar/drawer.tsx

```diff
@@ -56,6 +52,50 @@ const Component = (
         }
     };
 
+    const handleKeyDown = useCallback(
+        (event: React.KeyboardEvent<HTMLDivElement>) => {
+            if (!show) return;
+
+            if (event.key === "Escape") {
+                event.preventDefault();
+                onClose?.();
+                // Return focus to the hamburger menu button
+                setTimeout(() => {
+                    const hamburgerButton = document.querySelector(
+                        '[data-testid="button__mobile-menu"]'
```

Comment:
hmm we shouldn't use `data-testid` as that's meant for testing. also this might not work in some Storybook pages as it'll always pick the first match

can we save a ref for the MobileMenuButton and pass that down to Drawer? something like

```
const mobileMenuRef = useRef<HTMLButtonElement>(null);

<MobileMenuButton
   ref={mobileMenuRef}
>

<Drawer
   mobileMenuRef={mobileMenuRef}
>
```

and in drawer, we can do this directly:
```
mobileMenuRef.current?.focus()
```